File: 2023/rust/day01/src/part01.rs
use std::fs;

pub fn solve() {
    //     let text = "1abc2
    // pqr3stu8vwx
    // a1b2c3d4e5f
    // treb7uchet";
    let text = fs::read_to_string("input.txt").unwrap();

    let sum = text
        .trim()
        .split('\n')
        .map(|s| {
            let v = s.matches(char::is_numeric).collect::<Vec<&str>>();
            (String::from(v[0]) + v[v.len() - 1])
                .parse::<i32>()
                .unwrap()
        })
        .fold(0, |acc, v| acc + v);

    println!("{}", sum); // 56049
}

File: 2023/ts/15/part01.ts
import fs from "fs";
import path from "path";

{
  // const text = `rn=1,cm-,qp=3,cm=2,qp-,pc=4,ot=9,ab=5,pc-,pc=6,ot=7`;

  const text = fs.readFileSync(path.join(__dirname, "./input.txt"), {
    encoding: "ascii",
  });

  const ss = text.trim().split(",");
  const sum = ss
    .map((s) =>
      Array.from(s).reduce((acc, v) => ((acc + v.charCodeAt(0)) * 17) % 256, 0)
    )
    .reduce((acc, v) => acc + v, 0);

  console.log(sum); // 516469
}
